base/range.jl

```diff
@@ -59,6 +59,8 @@ automatically such that there are `length` linearly spaced elements in the range
 If `step` and `stop` are provided and `length` is not, the overall range length will be computed
 automatically such that the elements are `step` spaced.
 
+For a simpler linearly spaced range-like object with less overhead, see [`LinRange`](@ref).
```

Comment:
```suggestion
Special care is taken to ensure intermediate values are computed rationally.  For a simpler linearly spaced range-like object with less overhead, see [`LinRange`](@ref).
```

Ok, maybe the "rationally" pun is a little too cute, but I think we should mention why `range` is slower in the first place.

Reply:
I'll wait and see if anyone has thoughts on the use of the word 'rationally' before I commit the suggestion, but I agree it's good to point out why.

Reply:
I edited my old comment with a demo, you could use that: https://github.com/JuliaLang/julia/issues/35022#issuecomment-595464775.

Some people will look at that and feel very strongly that `range` is the "rational" one. Clearly you look at code, and by that definition `LinSpace` is the "rational" one.

Reply:
> I edited my old comment with a demo, you could use that: #35022 (comment).

I think that might be a bit too verbose to put in the docstring for `range`. Perhaps it can go in the `LinRange` docstring? Anyone have feelings on that? 

> Clearly you look at code, and by that definition LinSpace is the "rational" one. Though yes I like the pun.

Are you referring to me? I also think `range` is the 'rational' one and it's quite nice that it deals with floating point errors like that. I just didn't know there was a lower overhead alternative I should be using if I expect things to constant fold, and I suspect many others don't know about this overhead either. For someone who has never looked at the implementation of `range`, I think it's somewhat natural to think "how complicated could it be? Of course this thing should constant fold", so I thought it'd be good to put a hint in the docs.